android/titanium/src/java/org/appcelerator/titanium/util/TiConvert.java

```diff
@@ -852,6 +852,63 @@ public static Date toDate(HashMap<String, Object> hashMap, String key)
 	{
 		return toDate(hashMap.get(key));
 	}
+
+	public static final int MAX_LEVELS = 5;
+	public static final int MAX_SERLENGTH = 1000;
+	public static final int MAX_KEYS = 25;
+	public static final int MAX_KEYLENGTH = 50;
+
+	public static int validateJSON(JSONObject jsonObject, int level) {
+
+	    if (level > MAX_LEVELS) {
+	        Log.w(TAG, "Feature event cannot have more than "+ MAX_LEVELS + " nested JSONs");
+	        return TiC.ERROR_CODE_UNKNOWN;
+	    }
+	    if (jsonObject == null) {
+	        return TiC.ERROR_CODE_UNKNOWN;
+	    }
+	    if (level == 0 & jsonObject.toString().getBytes().length > MAX_SERLENGTH) {
```

Comment:
`if ((level == 0) & (jsonObject.toString().getBytes().length > MAX_SERLENGTH))`
Good to explicitly use braces when evaluating more than one expression. Improves readability.